File: problems/1394-lucky-number-in-an-array/main.js
const findLucky = function(arr) {
  const map = {};
  for (let i = 0; i < arr.length; i++) {
    if (map.hasOwnProperty(arr[i])) {
      map[arr[i]] = map[arr[i]] + 1;
    } else {
      map[arr[i]] = 1;
    }
  }
  let answer = -1;
  for (let obj in map) {
    if (parseInt(obj, 10) === map[obj]) {
      answer = map[obj];
    }
  }
  return answer;
};

File: problems/1394-lucky-number-in-an-array/main.cpp
class Solution {
  public:
    int findLucky(vector < int > & arr) {
      map < int, int > mp;
      for (int i: arr) {
        ++mp[i];
      }
      int ans = -1;
      for (auto m: mp) {
        if (m.first == m.second) {
          ans = m.first;
        }
      }
      return ans;
    }
};
